src/Symfony/Component/HttpFoundation/InputBag.php

```diff
@@ -35,8 +35,8 @@ public function get(string $key, $default = null)
 
         $value = parent::get($key, $this);
 
-        if (null !== $value && $this !== $value && !\is_scalar($value) && !(\is_object($value) && method_exists($value, '__toString'))) {
```

Comment:
the `__toString()` existence check is still relevant as it means the value can be casted to string despite `is_scalar()` doesn't recognize it as such

Reply:
True - but that check is missing in 6.x. So either the check in 5.4 needs to match the one in 6.x - or vice versa.

Reply:
I'd say it's a bug on 6.x then, given there's a `instanceof \Stringable` check (equivalent to the `__toString()` existence check as 6.x requires PHP 8) for `$default` above as well as in `InputBag::set()`

Reply:
Correct 👍